jina/serve/runtimes/worker/batch_queue.py

```diff
@@ -75,58 +76,64 @@ async def push(self, request: DataRequest) -> Task:
 
         :return: The task that will be set once the request has been processed.
         """
-        if not self._flush_task:
-            self._flush_task = asyncio.create_task(self._await_then_flush())
-        if not self._timer_task:
-            self._start_timer()
-
         docs = request.docs
-        # TODO: is this protected/synchronous ?
-        self._big_doc.extend(docs)
-        self._requests.append(request)
-        self._request_lens.append(len(docs))
 
-        if len(self._big_doc) >= self._preferred_batch_size:
-            self._flush_trigger.set()
+        # writes to shared data between tasks need to be mutually exclusive
+        async with self._data_lock:
```

Comment:
does it need to protect also the task creation? and the start timer?

Reply:
in my opinion yes, the other block calls reset which also sets the task to None